core/src/main/java/org/apache/accumulo/core/client/impl/TableOperationsImpl.java

```diff
@@ -1209,6 +1209,15 @@ public void online(String tableName) throws AccumuloSecurityException, AccumuloE
   public void online(String tableName, boolean wait) throws AccumuloSecurityException, AccumuloException, TableNotFoundException {
     checkArgument(tableName != null, "tableName is null");
     String tableId = Tables.getTableId(context.getInstance(), tableName);
+
+    /**
+     * ACCUMULO-4574 if table is already online return without executing fate operation.
+     */
+    TableState expectedState = Tables.getTableState(context.getInstance(), tableId);
```

Comment:
Why not clear the zoocache then? Further, why not clear that individual path to avoid impacting every other thread?